src/tallies/tally_scoring.cpp

```diff
@@ -2319,8 +2319,7 @@ void score_collision_tally(Particle& p)
 
 void score_surface_tally(Particle& p, const vector<int>& tallies)
 {
-  // No collision, so no weight change when survival biasing
-  double flux = p.wgt();
+  double current = p.wgt_last();
```

Comment:
This is because the weight might be adjusted in the case of an albedo or vacuum BC before the surface tally is run, correct? And you want to use the pre-crossing weight.

Just checking that I understand why this is changed to `wgt_last`.

Reply:
Exactly -- when we have vacuum BCs, we set the weight to zero but then score surface tallies afterward. Before, we only relied on `alive_` and the weight was never changed when hitting a vacuum BC.